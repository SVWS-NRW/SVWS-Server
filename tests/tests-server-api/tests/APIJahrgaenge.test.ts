import {describe, expect, test} from "vitest";
import {getApiServer} from "./utils/TestUtils.js";

describe("Jahrgaenge Tests", () => {
	describe.each([{schema: "GymAbi01"}])('gegen %s', ({schema}) => {
		const api = getApiServer(schema);

		test("getJahrgaenge", async () => {
			const result = await api.getJahrgaenge(schema);
			expect(result).toMatchSnapshot();
		});

		test("getJahrgang", async () => {
			const result = await api.getJahrgang(schema, 10);
			expect(result).toMatchSnapshot();
		});
	})
})
