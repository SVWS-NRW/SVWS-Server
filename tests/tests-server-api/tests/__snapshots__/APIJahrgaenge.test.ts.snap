// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Jahrgaenge Tests > gegen { schema: 'GymAbi01' } > getJahrgaenge 1`] = `
ArrayList {
  "addFirst": [Function],
  "addLast": [Function],
  "elementData": [
    JahrgangsDaten {
      "anzahlRestabschnitte": 16,
      "bezeichnung": "  5. Jahrgang",
      "gueltigBis": null,
      "gueltigVon": null,
      "id": 1,
      "idFolgejahrgang": 2,
      "istSichtbar": true,
      "kuerzel": "05",
      "kuerzelSchulgliederung": "***",
      "kuerzelStatistik": "05",
      "kurzbezeichnung": null,
      "referenziertInAnderenTabellen": null,
      "sortierung": 10,
    },
    JahrgangsDaten {
      "anzahlRestabschnitte": 14,
      "bezeichnung": "  6. Jahrgang",
      "gueltigBis": null,
      "gueltigVon": null,
      "id": 2,
      "idFolgejahrgang": 3,
      "istSichtbar": true,
      "kuerzel": "06",
      "kuerzelSchulgliederung": "***",
      "kuerzelStatistik": "06",
      "kurzbezeichnung": null,
      "referenziertInAnderenTabellen": null,
      "sortierung": 20,
    },
    JahrgangsDaten {
      "anzahlRestabschnitte": 12,
      "bezeichnung": "  7. Jahrgang",
      "gueltigBis": null,
      "gueltigVon": null,
      "id": 3,
      "idFolgejahrgang": 4,
      "istSichtbar": true,
      "kuerzel": "07",
      "kuerzelSchulgliederung": "***",
      "kuerzelStatistik": "07",
      "kurzbezeichnung": null,
      "referenziertInAnderenTabellen": null,
      "sortierung": 30,
    },
    JahrgangsDaten {
      "anzahlRestabschnitte": 10,
      "bezeichnung": "  8. Jahrgang",
      "gueltigBis": null,
      "gueltigVon": null,
      "id": 4,
      "idFolgejahrgang": 5,
      "istSichtbar": true,
      "kuerzel": "08",
      "kuerzelSchulgliederung": "***",
      "kuerzelStatistik": "08",
      "kurzbezeichnung": null,
      "referenziertInAnderenTabellen": null,
      "sortierung": 40,
    },
    JahrgangsDaten {
      "anzahlRestabschnitte": 8,
      "bezeichnung": "  9. Jahrgang",
      "gueltigBis": null,
      "gueltigVon": null,
      "id": 5,
      "idFolgejahrgang": 10,
      "istSichtbar": true,
      "kuerzel": "09",
      "kuerzelSchulgliederung": "***",
      "kuerzelStatistik": "09",
      "kurzbezeichnung": null,
      "referenziertInAnderenTabellen": null,
      "sortierung": 50,
    },
    JahrgangsDaten {
      "anzahlRestabschnitte": 6,
      "bezeichnung": "Einführungsphase",
      "gueltigBis": null,
      "gueltigVon": null,
      "id": 10,
      "idFolgejahrgang": 11,
      "istSichtbar": true,
      "kuerzel": "EF",
      "kuerzelSchulgliederung": "***",
      "kuerzelStatistik": "EF",
      "kurzbezeichnung": null,
      "referenziertInAnderenTabellen": null,
      "sortierung": 60,
    },
    JahrgangsDaten {
      "anzahlRestabschnitte": 4,
      "bezeichnung": "Qualifikationsphase 1",
      "gueltigBis": null,
      "gueltigVon": null,
      "id": 11,
      "idFolgejahrgang": 12,
      "istSichtbar": true,
      "kuerzel": "Q1",
      "kuerzelSchulgliederung": "***",
      "kuerzelStatistik": "Q1",
      "kurzbezeichnung": null,
      "referenziertInAnderenTabellen": null,
      "sortierung": 70,
    },
    JahrgangsDaten {
      "anzahlRestabschnitte": 2,
      "bezeichnung": "Qualifikationsphase 2",
      "gueltigBis": null,
      "gueltigVon": null,
      "id": 12,
      "idFolgejahrgang": null,
      "istSichtbar": true,
      "kuerzel": "Q2",
      "kuerzelSchulgliederung": "***",
      "kuerzelStatistik": "Q2",
      "kurzbezeichnung": null,
      "referenziertInAnderenTabellen": null,
      "sortierung": 80,
    },
  ],
  "getFirst": [Function],
  "getLast": [Function],
  "modCount": 8,
  "removeFirst": [Function],
  "removeLast": [Function],
}
`;

exports[`Jahrgaenge Tests > gegen { schema: 'GymAbi01' } > getJahrgang 1`] = `
JahrgangsDaten {
  "anzahlRestabschnitte": 6,
  "bezeichnung": "Einführungsphase",
  "gueltigBis": null,
  "gueltigVon": null,
  "id": 10,
  "idFolgejahrgang": 11,
  "istSichtbar": true,
  "kuerzel": "EF",
  "kuerzelSchulgliederung": "***",
  "kuerzelStatistik": "EF",
  "kurzbezeichnung": "EF",
  "referenziertInAnderenTabellen": null,
  "sortierung": 60,
}
`;
